pkg/plugin/framer.go

```diff
@@ -52,3 +55,70 @@ func ConvertStatsV2ResponseToFrame(frameName string, stats sentry.StatsV2Respons
 	}
 	return frame, nil
 }
+
+type SentryEventsStatsSet struct {
+	Name string
+	Data []interface{}
+}
+
+func ConvertEventStatsSetToTimestampField(set SentryEventsStatsSet) (*data.Field){
+	field := data.NewFieldFromFieldType(data.FieldTypeTime, len(set.Data))
+	field.Name = "Timestamp"
+	for index, value := range set.Data {
+		row := value.([]interface{})
+		field.Set(index, time.Unix(int64(row[0].(float64)), 0))
+	}
+	return field
+}
+
+func ConvertEventStatsSetToField(set SentryEventsStatsSet) (*data.Field){
+	field := data.NewFieldFromFieldType(data.FieldTypeNullableFloat64, len(set.Data))
+	field.Name = set.Name
+	for index, value := range set.Data {
+		row := value.([]interface{})
+		rawCount := row[1].(([]interface{}))[0].(map[string]interface{})["count"]
```

Comment:
I’m under the impression that if these casts are bad it would panic, which I think is a reasonable behavior. If sentry changes their schema I can’t really recover gracefully with type assertions, except maybe with an empty response (which I think is worse).

Reply:
> I’m under the impression that if these casts are bad it would panic, which I think is a reasonable behavior.

Generally, it is convention within our datasources that (where possible) the error is attached to the data frame and returned to the client rather than `panic`ing and letting the process die. This function should probably return an instance of `error` that is then handled at the top-level within the `QueryData` handler.